### PR TITLE
HLT Module for Saving DIGIs of Rechits for Pi0/Eta Calibration Stream [74X]

### PR DESCRIPTION
**Pull request by hardenbr**

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+};
+//
+
+//
+// constructors and destructor
+//
+HLTRechitsToDigis::HLTRechitsToDigis(const edm::ParameterSet& iConfig)
+{
+  //region to do rechit digi matching
+  regionID_ = iConfig.getParameter<uint32_t> ("regionID_1EB_2EE");
```
Comment (fwyzard) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: No, just no. Please use simply "region" as the parameter name.

You can use strings ("EE" / "EB" or "endcap" / "barrel" or whatever you prefer) and covert them integers (better, an enum) in a helper function.
Or, plain integers as you are using, and add a comment in the cfi file explaining the meaning.

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+*/
+ 
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+HLTRechitsToDigis::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+
+  desc.add<uint32_t>("regionID_1EB_2EE", 1);
```
Comment (fwyzard) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: please put a description of the possible parameters in the comments for the parameter.

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+// class declaration
+//
+
+class HLTRechitsToDigis : public edm::stream::EDProducer<> {
+public:
+  explicit HLTRechitsToDigis(const edm::ParameterSet&);
+  ~HLTRechitsToDigis();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  enum ecalRegion {invalidRegion=0, barrel, endcap };  
+  HLTRechitsToDigis::ecalRegion stringToRegion(std::string region);
```
Comment (fwyzard) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: this should be a `static` method

Comment (fwyzard) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: and you should pass `region` by const reference, not by value.

File: HLTrigger/special/src/HLTRechitsToDigis.cc [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+{ 
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+HLTRechitsToDigis::ecalRegion
+HLTRechitsToDigis::stringToRegion(const std::string & region) {
```
Comment (Martin-Grunewald) [inline HLTrigger/special/src/HLTRechitsToDigis.cc]: Should be static and should be const.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**